srtcore/core.cpp

```diff
@@ -10029,7 +10029,7 @@ int srt::CUDT::processData(CUnit* in_unit)
                 continue;
             }
 
-            const int avail_bufsize = getAvailRcvBufferSizeNoLock();
+            const long avail_bufsize = getAvailRcvBufferSizeNoLock();
```

Comment:
I think you better use `size_t`, as this is the original type returned by the called function, then the below condition should be:
```
if (size_t(offset) >= avail_bufsize)
```
This is logically better because here you definitely compare a positive value of `offset`, as the negative one has been handled already earlier. This should work well, as `offset` is `int32_t`, already known to be positive, so it will be extended to 64 bits with 0 bits.

Reply:
Sounds good to me, done.